clients/admin-ui/src/features/privacy-requests/buttons/ReprocessButton.tsx

```diff
@@ -0,0 +1,113 @@
+import { Box, Button, ButtonProps, forwardRef, Text } from "@fidesui/react";
+import { useState } from "react";
+
+import { useAppDispatch, useAppSelector } from "~/app/hooks";
+import { useAlert, useAPIHelper } from "~/features/common/hooks";
+
+import {
+  selectRetryRequests,
+  setRetryRequests,
+  useBulkRetryMutation,
+  useRetryMutation,
+} from "../privacy-requests.slice";
+import { PrivacyRequest } from "../types";
+
+type ReprocessButtonProps = {
+  buttonProps?: ButtonProps;
+  subjectRequest?: PrivacyRequest;
+};
+
+const ReprocessButton = forwardRef(
+  ({ buttonProps, subjectRequest }: ReprocessButtonProps, ref) => {
+    const dispatch = useAppDispatch();
+    const [isReprocessing, setIsReprocessing] = useState(false);
+    const { handleError } = useAPIHelper();
+    const { errorAlert, successAlert } = useAlert();
+
+    const { errorRequests } = useAppSelector(selectRetryRequests);
+    const [bulkRetry] = useBulkRetryMutation();
+    const [retry] = useRetryMutation();
+
+    const handleBulkReprocessClick = async () => {
+      setIsReprocessing(true);
+      try {
+        const payload = await bulkRetry(errorRequests).unwrap();
```

Comment:
[Still curious](https://github.com/ethyca/fides/pull/1489#discussion_r1001287396) why this is an `unwrap`. Using the RTK error handling is [pretty clean](https://github.com/ethyca/fides/blob/main/clients/admin-ui/src/features/dataset/DatabaseConnectForm.tsx#L84).

Reply:
I want direct access to the success or error payload immediately after the mutation. In my opinion, this is cleaner using a try/catch which is more legible rather than having another if statement to detect a string literal "error" in the mutation response. Different approach. Thanks for sharing.

Reply:
I have to disagree. RTK Query  provides [type safe error handling](https://redux-toolkit.js.org/rtk-query/usage-with-typescript#type-safe-error-handling). The the error property from the non-unwrapped result has more information than what `catch` gives you, which is why you have [to cast](https://github.com/ethyca/fides/pull/1489/files/97dee204413a06089b7b9dc624892988c8dd0a9c#diff-1d10f174b815bd14f5a2ec131cffdbd469495c8b76cdf9b1d47d4acc2f8a1541R55) `error as string`. (Also, that cast has a chance of introducing a bug. A thrown error can be anything. Coercing the error to a string is safer: `String(error)` or  ``` `${error}` ```)